### DISCUSSION TITLE
Can ruff implement generic text rules and fixes?

### DISCUSSION THREAD

**Discussion by ssbarnea** [astral-sh/ruff#4150] [General]
There is range of generic fixes that are not unique to python files, likely applicable to any text file, such

- fix end of files
- trim trailing whitespace
- mixed line ending
- BOMs (byte-order-mask)
- check that executables have shebangs
- check for merge conflicts
- debug statements

At this moment we are using [pre-commit hooks](https://github.com/pre-commit/pre-commit-hooks.git) for each of these, but I am wondering if it would be desirable to add these features to ruff, for speed and simplicity.

**Comment by Avasam** [comment]
You could consider dprint for non-python files: https://dprint.dev/config/#global-configuration It's a multi-language formatter also written in Rust !
